examples/trigger/trigger.go

```diff
@@ -0,0 +1,175 @@
+		reportNodeID  = flag.String("report", "", "node id value to report on trigger")
+		filter        = flag.String("filter", "timestamp", "DataFilter: status, value, timestamp.")
+		interval      = flag.String("interval", opcua.DefaultSubscriptionInterval.String(), "subscription interval")
+	)
+	flag.BoolVar(&debug.Enable, "debug", false, "enable debug logging")
+	flag.Parse()
+	log.SetFlags(0)
+
+	subInterval, err := time.ParseDuration(*interval)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// add an arbitrary timeout to demonstrate how to stop a subscription
+	// with a context.
+	d := 60 * time.Second
+	ctx, cancel := context.WithTimeout(context.Background(), d)
+	defer cancel()
+
+	endpoints, err := opcua.GetEndpoints(*endpoint)
+	if err != nil {
+		log.Fatal(err)
+	}
+	ep := opcua.SelectEndpoint(endpoints, *policy, ua.MessageSecurityModeFromString(*mode))
+	if ep == nil {
+		log.Fatal("Failed to find suitable endpoint")
+	}
+
+	fmt.Println("*", ep.SecurityPolicyURI, ep.SecurityMode)
+
+	opts := []opcua.Option{
+		opcua.SecurityPolicy(*policy),
+		opcua.SecurityModeString(*mode),
+		opcua.CertificateFile(*certFile),
+		opcua.PrivateKeyFile(*keyFile),
+		opcua.AuthAnonymous(),
+		opcua.SecurityFromEndpoint(ep, ua.UserTokenTypeAnonymous),
+	}
+
+	c := opcua.NewClient(ep.EndpointURL, opts...)
+	if err := c.Connect(ctx); err != nil {
+		log.Fatal(err)
+	}
+	defer c.Close()
+
+	notifyCh := make(chan *opcua.PublishNotificationData)
+
+	sub, err := c.Subscribe(&opcua.SubscriptionParameters{
+		Interval: subInterval,
+	}, notifyCh)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer sub.Cancel()
+	log.Printf("Created subscription with id %v", sub.SubscriptionID)
+
+	triggeringNode, err := ua.ParseNodeID(*triggerNodeID)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	triggeredNode, err := ua.ParseNodeID(*reportNodeID)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	miCreateRequests := []*ua.MonitoredItemCreateRequest{
+		opcua.NewMonitoredItemCreateRequestWithDefaults(triggeringNode, ua.AttributeIDValue, 42),
+		{
+			ItemToMonitor: &ua.ReadValueID{
+				NodeID:       triggeredNode,
+				AttributeID:  ua.AttributeIDValue,
+				DataEncoding: &ua.QualifiedName{},
+			},
+			MonitoringMode: ua.MonitoringModeSampling,
+			RequestedParameters: &ua.MonitoringParameters{
+				ClientHandle:     43,
+				DiscardOldest:    true,
+				Filter:           getFilter(*filter),
+				QueueSize:        10,
+				SamplingInterval: 0.0,
+			},
+		},
+	}
+
+	subRes, err := sub.Monitor(ua.TimestampsToReturnBoth, miCreateRequests...)
+	if err != nil || subRes.Results[0].StatusCode != ua.StatusOK {
+		log.Fatal(err)
+	}
+
+	go sub.Run(ctx) // start Publish loop
+
+	triggeringServerID, triggeredServerID := subRes.Results[0].MonitoredItemID, subRes.Results[1].MonitoredItemID
+	tRes, err := sub.SetTriggering(triggeringServerID, []uint32{triggeredServerID}, []uint32{})
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Print(tRes)
```

Comment:
This just dumps `&{0xc00020f6d0 [The monitoring item id does not refer to a valid monitored item. StatusBadMonitoredItemIDInvalid (0x80420000)] [] [] []}`. Is that useful?

Reply:
i changed it to check the status and throw an error if not ok instead.